closure/goog/useragent/flash.js

```diff
@@ -60,49 +60,55 @@ goog.userAgent.flash.init_ = function() {
       if (plugin.description) {
         goog.userAgent.flash.detectedFlashVersion_ =
             goog.userAgent.flash.getVersion_(plugin.description);
+        return;
       }
     }
 
     if (navigator.plugins['Shockwave Flash 2.0']) {
       goog.userAgent.flash.detectedFlash_ = true;
       goog.userAgent.flash.detectedFlashVersion_ = '2.0.0.11';
+      return;
     }
+  }
 
-  } else if (navigator.mimeTypes && navigator.mimeTypes.length) {
+  if (navigator.mimeTypes && navigator.mimeTypes.length) {
     var mimeType = navigator.mimeTypes['application/x-shockwave-flash'];
     goog.userAgent.flash.detectedFlash_ = mimeType && mimeType.enabledPlugin;
     if (goog.userAgent.flash.detectedFlash_) {
       goog.userAgent.flash.detectedFlashVersion_ =
           goog.userAgent.flash.getVersion_(mimeType.enabledPlugin.description);
+      return;
     }
+  }
 
-  } else {
+  /** @preserveTry */
+  try {
+    // Try 7 first, since we know we can use GetVariable with it
+    var ax = new ActiveXObject('ShockwaveFlash.ShockwaveFlash.7');
+    goog.userAgent.flash.detectedFlash_ = true;
+    goog.userAgent.flash.detectedFlashVersion_ =
+        goog.userAgent.flash.getVersion_(ax.GetVariable('$version'));
+    return;
+  } catch (e) {
```

Comment:
This should be simply

    } catch (e) { /* fall-through */ }

Since you're not using the exceptions, there's no need to do the work inside the `catch` block, and introduce an extra level of indentation and require new names `e2`, and `e3`.